src/main/java/backend/control/RepoOpControl.java

```diff
@@ -42,6 +64,26 @@ public RepoOpControl(RepoIO repoIO) {
         return result;
     }
 
+    /**
+     * Updates repository stored locally with data from a GitHubModelUpdatesData object.
+     * Set syncOperation to queue this operation in the blocking queue for the updating repository
+     * @param updates
+     * @param syncOperation
+     * @return
+     */
+    public CompletableFuture<Model> updateLocalModel(GitHubModelUpdatesData updates,
+                                                     boolean syncOperation) {
+        init(updates.getRepoId());
+        CompletableFuture<Model> result = new CompletableFuture<>();
+        UpdateLocalModelOp op = new UpdateLocalModelOp(models, updates, result);
+        if (syncOperation) {
```

Comment:
I see. That's tough. I guess the recursive updating would need to be implemented by having the current task fail, then making whatever is calling it put a new task on the queue.

This recursiveness is the reason for having to synchronise only sometimes, right? We may enter another updateModel and don't want to enqueue it in that case. I think the refactoring you suggested is probably the way to go for the long term, but I'm not sure if we want to do it now.